### ISSUE TITLE
[Feature request] ldc2 support

### ISSUE THREAD

**Issue by Dadoum** [andrewlalis/slf4d#3]
it currently makes the following error:
```
 ../../../.dub/packages/slf4d-2.4.2/slf4d/source/slf4d/default_provider/factory.d(34,22): Error: non-shared method `core.sync.rwmutex.ReadWriteMutex.this` is not callable using a `shared` object
/usr/lib/ldc/x86_64-linux-gnu/include/d/core/sync/rwmutex.d(89,5):        Consider adding `shared` here
../../../.dub/packages/slf4d-2.4.2/slf4d/source/slf4d/default_provider/factory.d(43,22): Error: non-shared method `core.sync.rwmutex.ReadWriteMutex.writer` is not callable using a `shared` object
/usr/lib/ldc/x86_64-linux-gnu/include/d/core/sync/rwmutex.d(148,22):        Consider adding `shared` here
../../../.dub/packages/slf4d-2.4.2/slf4d/source/slf4d/default_provider/factory.d(63,22): Error: non-shared method `core.sync.rwmutex.ReadWriteMutex.writer` is not callable using a `shared` object
/usr/lib/ldc/x86_64-linux-gnu/include/d/core/sync/rwmutex.d(148,22):        Consider adding `shared` here
../../../.dub/packages/slf4d-2.4.2/slf4d/source/slf4d/default_provider/factory.d(96,22): Error: non-shared method `core.sync.rwmutex.ReadWriteMutex.reader` is not callable using a `shared` object
/usr/lib/ldc/x86_64-linux-gnu/include/d/core/sync/rwmutex.d(136,22):        Consider adding `shared` here
ldc2 failed with exit code 1.
```
edit: that's related to the ldc2 version.

**Comment by andrewlalis** [comment]
Just for reference, can you please provide the version of LDC2 which caused this conflict? @Dadoum

**Comment by Dadoum** [comment]
1.28.0, which is the one shipped by Ubuntu 22.04 LTS repos
